From 48b3238c50314d0a2a5ab62c03868c9dedd7a360 Mon Sep 17 00:00:00 2001
From: Wolfgang Bauer <wbauer@tmo.at>
Date: Thu, 23 Apr 2020 18:56:39 +0200
Subject: [PATCH] Fix build with Qt 5.9

`qsizetype` is only defined starting with Qt 5.10.
To avoid a compiler error, conditionally define it appropriately for
older Qt versions (the actual line is copied from Qt 5.10's qglobal.h).

Differential Revision: https://phabricator.kde.org/D29134
---
 src/images/image.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/images/image.h b/src/images/image.h
index eb287db9..3081b831 100644
--- a/src/images/image.h
+++ b/src/images/image.h
@@ -30,6 +30,10 @@
 #include <QByteArray>
 #include <QPixmap>
 
+#if (QT_VERSION < QT_VERSION_CHECK(5, 10, 0))
+using qsizetype = QIntegerForSizeof<std::size_t>::Signed;
+#endif
+
 namespace Tellico {
   class ImageFactory;
   class ImageDirectory;
-- 
2.16.4

